src/zarr/__init__.py

```diff
@@ -36,6 +36,25 @@
 # in case setuptools scm screw up and find version to be 0.0.0
 assert not __version__.startswith("0.0.0")
 
+
+def print_debug_info() -> None:
+    """
+    Print version info for use in bug reports.
+    """
+    import platform
+    from importlib import import_module
+
+    print(f"platform: {platform.platform()}")
+    print(f"python: {platform.python_version()}\n")
+
+    print(f"zarr: {__version__}\n")
+    for package in ["numcodecs", "numpy", "fsspec", "s3fs", "botocore", "gcsfs"]:
```

Comment:
can we include the full list of required and optional dependencies here? 

Required dependencies: donfig | numcodecs | numpy | packaging | typing-extensions
Optional dependencies: botocore | cupy-cuda12x | fsspec| numcodecs | rich | s3fs | universal-pathlib | obstore

Reply:
I added all of them. They're in a list because introspecting the package to grab it's own dependencies seemed to add a bit of unnecessary complication